pages/data-providers/index.jsx

```diff
@@ -49,9 +50,16 @@ const useDataProviderController = () => {
 const getHelperMessage = ({ created, duplicated, error }) => {
   if (error) {
     return {
-      message:
-        'We could not find any OAI-PMH endpoint.' +
-        ' Please try it with another URL.',
+      message: (
+        <>
+          We could not find any OAI-PMH endpoint. Please try it with another
+          URL. If you believe the URL is correct please contant us at{' '}
+          <Link href="mailto:thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k">
+            thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k
```

Comment:
This will not work so. You should obfuscate it using percent-encoding (URL-like). 

I used this (my custom) script:
```js
'me@example.com'.split('').map(c => Math.random() <= 0.4 || c == '.' || c == '@' ? `%${c.charCodeAt().toString(16).padStart(2, '0')}` : c).join('')
```

It replaces about 60% of characters to corresponding hex but dots and at are always replaced.

Reply:
Why do you think so? It's fine from chrome

![email](https://user-images.githubusercontent.com/12775440/89183654-d6dff180-d597-11ea-80a0-00b311348d92.PNG)
![email1](https://user-images.githubusercontent.com/12775440/89183689-e5c6a400-d597-11ea-9afb-8e7a4889ade9.PNG)

Reply:
I was cached by this when was doing this previously. The actual URL should be percent-decoded and HTML part should be in HTML entities.

It surprisingly works because of the way browsers interpret HTML code. Here is my step-by-step exploration.

1️⃣ I checked how it works and it's fine in all browsers (Safari, Firefox and Chrome). However, I assume this works because React or Babel decodes this at some point (everywhere I see the decoded address and text for some reason, in the React Components tool and pure HTML elements tool). This is not expected behaviour though and I am still wondering.

2️⃣ So, I went to further exploration. If you open the following page:

```html
<a href="mailto:thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k">
  thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k
</a>

<br>

<script>
  const a = document.createElement('a')
  a.setAttribute('href', 'mailto:thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k')
  a.innerHTML = 'thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k'
  document.body.appendChild(a)
</script>
```

The first link will work when the second won't. This is because the HTML engine processes entities but when you add DOM through JS, attributes are not parsed.

3️⃣ However, when you run slightly different snippet:

```html
<p>Link is inserted here</p>

<script>
  const p = document.querySelector('p')
  p.innerHTML = `
    <a href="mailto:thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k">
      thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k
    </a>
  `
</script>
```

everything works as expected because of HTML engine processing the string passed to `innerHTML` and fixing errors.

4️⃣ When you finally do something like this in pure HTML:

```html
<a href="mailto:the%74ea%6d%40c%6f%72%65%2eac%2e%75%6b">
    thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k
</a>
```
you will find that the attribute is passed unchanged straight and is decoded only after the click.

It does not mean we need the final effect but I think it's safer to process the address in this way if you change the code but not the data.

I was passing the obfuscated address to Markdown so it was not processed and the HTML entities in the attribute leaked to the rendered DOM were breaking the behaviour.

Unfortunately, there is no easy way to replace chars in the protocol prefix but this looks safe enough to me. At least some crawlers will not collect this.

---

Off-topic, I think we should have somewhere the registry (an object, a module or in the context) with all the obfuscated addresses.

Reply:
Thanks for a thorough explanation. It was really interesting to read. So I will keep it as is then.


I am just wondering what encoding is used for `mailto:thet&#101;&#97;m&#64;c&#111;re&#46;&#97;c&#46;&#117;k`?

Reply:
UTF-8, or did you mean something else?

I would rather convert this to URL percentage formatting because if you copy the address to let's say Markdown, it will stop working and nobody notices. However, since you already merged, it's fine.